src/js/controllers/mapController.ts

```diff
@@ -402,6 +417,227 @@ export class MapController {
+    switch (optionType) {
+      case 'area':
+        this._selectedWidget = new AreaMeasurement2D({
+          view: this._mapview,
+          unit: 'acres'
+        });
+        break;
+      case 'distance':
+        this._selectedWidget = new DistanceMeasurement2D({
+          view: this._mapview,
+          unit: 'miles'
+        });
+        break;
+      case 'coordinates': {
+        this._selectedWidget?.viewModel.clearMeasurement();
+        this._selectedWidget = undefined;
+        // this.updateOnClickCoordinates(selectedDropdownOption);
+        // this.setOnClickCoordinates(selectedDropdownOption);
+        // this.setPointerMoveCoordinates(selectedDropdownOption);
+        break;
+      }
+      default:
+        break;
+    }
+
+    if (optionType === 'area' || optionType === 'distance') {
+      this._selectedWidget?.viewModel.newMeasurement();
+      this.getAndDispatchMeasureResults(optionType);
+    }
+  }
+
+  updateSelectedMeasureWidget(
+    optionType: string,
+    selectedUnit: AreaMeasurement2D['unit'] | DistanceMeasurement2D['unit']
+  ): void {
+    let areaResults = {};
+    let distanceResults = {};
+
+    if (this._selectedWidget) {
+      this._selectedWidget.unit = selectedUnit;
+      switch (optionType) {
+        case 'area':
+          areaResults = {
+            area: this._selectedWidget.viewModel.measurementLabel.area,
+            perimeter: this._selectedWidget.viewModel.measurementLabel.perimeter
+          };
+          break;
+        case 'distance':
+          distanceResults = {
+            length: this._selectedWidget.viewModel.measurementLabel
+          };
+          break;
+        default:
+          break;
+      }
+
+      store.dispatch(
+        setMeasureResults({
+          areaResults,
+          distanceResults,
+          coordinateMouseClickResults: {},
+          coordinatePointerMoveResults: {}
+        })
+      );
+    }
+    this.updateMeasureWidgetOnClick();
+  }
+
+  updateOnClickCoordinates(selectedDropdownOption: string): void {
+    const {
+      coordinateMouseClickResults
+    } = store.getState().appState.measureContent.results;
+    const isDMS = selectedDropdownOption === 'dms';
+    const isDecimal = selectedDropdownOption === 'decimal';
+
+    if (
+      coordinateMouseClickResults?.latitude &&
+      coordinateMouseClickResults?.longitude &&
+      isDMS
+    ) {
+      // TODO - convert decimal to DMS
+      // * NOTE - Will need to revisit this logic
+      // * NOTE - Will need to explicitly update other ...Results property of Redux state
+
+      store.dispatch(
+        setMeasureResults({
+          areaResults: {},
+          distanceResults: {},
+          coordinateMouseClickResults: {}
+        })
+      );
+    } else if (
+      coordinateMouseClickResults?.latitude &&
+      coordinateMouseClickResults?.longitude &&
+      isDecimal
+    ) {
+      // TODO - convert DMS to decimal
+    }
+  }
+
+  updateMeasureWidgetOnClick(): void {
```

Comment:
Bug": 🐛1. Area widget, draw an area, do not double click. 2) select unit from dropdown 3) hover over the map and "click" 4) drawing disappears! . Looks like we cleared the map widget content when we did not need to.

Reply:
this PR!